lib/ical_filter_proxy/calendar_builder.rb

```diff
@@ -33,5 +34,37 @@ def add_rules
       end
     end
 
+    def add_alarms
+      alarms = calendar_config["alarms"]
+      return unless alarms
+
+      if alarms["clear_existing"] == true
+        calendar.set_clear_existing_alarms
+      end
+
+      triggers = alarms["triggers"]
+      return unless triggers
+
+      triggers.each do |trigger|
+        calendar.add_trigger(generate_iso_format(trigger))
```

Comment:
Can we get this down to be as simple as `calendar.add_rule` above?

We could go about this by creating a new `AlarmTrigger` class similar to `FilterRule` and move the `generate_iso_format` logic into that too.

Reply:
Will do. Funnily I had it like that but than reduced it to just the triggers, but now makes sense again ;)

Reply:
Added it. Can you have a look again @darkphnx ?